### DISCUSSION TITLE
Spotify to iDrive 6 Bmw M4 2017...

### DISCUSSION THREAD

**Discussion by Esco54** [BimmerGestalt/AAIdrive#260] [Q&A]
Hi, was reading on reddit about a solution to connect Spotify again to the iDrive 6 screen? Can someone please help me with this issue? Thx!

**Comment by hufman** [comment]
Hello, and welcome! I hope we can get it working for you!
First, does your car's Bluetooth device settings have an Apps option?
![pic_btapp_bmw](https://user-images.githubusercontent.com/1592375/115940819-c0483600-a457-11eb-91d9-8ca92730136a.jpg)

Next, you have to get BMW Connected or MyBMW to connect to the car and check this box. The latest BMW Connected app [seems to no longer connect](https://github.com/hufman/AndroidAutoIdrive/discussions/238) to the car, so you'll want to find BMW Connected 6.4 from apkpure or something.

Once the car connection is verified to work, your phone is ready to install AndroidAutoIdrive. The release from August doesn't support MyBMW, so I recommend installing [the nightly build](https://androidautoidrive.s3.amazonaws.com/hufman/AndroidAutoIdrive/androidautoidrive-latest-master-nomap-sentry-release.apk). Once installed, it should start up and detect the car connection and begin working.

Please let us know if the [quickstart documentation](https://github.com/hufman/AndroidAutoIdrive#getting-started) is missing some information!

**Comment by Esco54** [comment]
Where can i download AndroidAutoIdrive? Thx!

**Reply by hufman** [reply]
You can find all of the major releases on the [Releases page](https://github.com/hufman/AndroidAutoIdrive/releases), as well as a [direct link](https://androidautoidrive.s3.amazonaws.com/hufman/AndroidAutoIdrive/androidautoidrive-latest-master-nomap-sentry-release.apk) to the automatic build from each code push. Just download the APK file and install it on your Android handset.

**Comment by Esco54** [comment]
Hi,

I just switched to a Bmw M4 with Idrive 4...Can i use the Spotify app on it?
Thx! :)

Kurt

Outlook voor Android<https://aka.ms/AAb9ysg> downloaden
________________________________

**Reply by hufman** [reply]
Sure, depending on whether your car has Apps support. Does it have the Connection Assistant under the ConnectedDrive menu? https://github.com/hufman/AndroidAutoIdrive/blob/master/app/src/main/res/drawable/pic_connassistant_bmw.jpg?raw=true
IDrive4 only supports USB connections, which the new MyBMW app doesn't seem to support, so you'll need to use BMW Connected 6.4.
Because of this USB connection, the official Spotify app on Android was never supported in IDrive4, so AAI is the best way to integrate with it.

**Comment by Esco54** [comment]
What do you exactly mean by only USB connections?

Outlook voor Android<https://aka.ms/AAb9ysg> downloaden

________________________________

**Reply by hufman** [reply]
IDrive4 can only run apps through USB, while IDrive 5 and 6 can run apps over Bluetooth. Your car will listen to your phone over Bluetooth audio, and AAI will provide a interface to control the music apps or other features through the USB connection to the car.

**Comment by Esco54** [comment]
How can i know if i have Idrive 4 or 5?
Thanks :)

Outlook voor Android<https://aka.ms/AAb9ysg> downloaden
________________________________

**Reply by hufman** [reply]
If you have a Connection Wizard under your Connected Drive menu, you have ID4. If you have a Mobile Devices menu where a phone can have an Apps option besides the Phone and Music, you have ID5.

**Comment by Esco54** [comment]
Hi Walter,

Can the 6NR option Bmw apps be coded in to my 2015 M4?

Outlook voor Android<https://aka.ms/AAb9ysg> downloaden
________________________________

**Reply by hufman** [reply]
I assume so, but I'm unfamiliar with coding and can't say for certain what options are available.
